modules/meta.analysis/R/single.MA.R

```diff
@@ -18,106 +19,107 @@
-                             inits = j.inits,
-                             n.chains = j.chains)
-
-    jags.out   <- coda.samples ( model = j.model,
-                                variable.names = vars,
-                                n.iter = j.iter,
-                                thin = max(c(2,j.iter/(5000*2))))
-    ## I would have done a while loop, but it could take forever
-    ## So just give one chance to try again
-    if(gelman.diag(jags.out)$mpsrf > 1.2){
-      logger.warn("model did not converge; re-running with j.iter * 10")
-      jags.out   <- coda.samples ( model = j.model,
-                                   variable.names = vars,
-                                   n.iter = j.iter * 10,
-                                   thin = max(c(2,j.iter/(500*2))))
+  }
+  
+  ### Import defaul JAGS model file
+  modelfile <- system.file("ma.model.template.bug", package = "PEcAn.MA")
+  
+  ### Write JAGS bug file based on user settings and default bug file write.ma.model (modelfile =
+  ### paste(settings$pecanDir,'rscripts/ma.model.template.bug',sep=''),
+  write.ma.model(modelfile = modelfile, 
+                 outfile = jag.model.file, 
+                 reg.model = reg.model, 
+                 jagsprior$distn, jagsprior$a, jagsprior$b, 
+                 n = length(data$Y), 
+                 trt.n = model.parms[["trt"]],
+                 site.n = model.parms[["site"]], 
+                 ghs.n = model.parms[["ghs"]], 
+                 tauA = tauA, tauB = tauB)
+  
+  if (overdispersed == TRUE) {
+    ## overdispersed chains
+    j.inits <- function(chain) {
+      list(beta.o = do.call(paste("q", prior$dist, sep = ""), 
+                            list(chain * 1 / (j.chains + 1), prior$a, prior$b)), 
+           .RNG.seed = chain, 
+           .RNG.name = "base::Mersenne-Twister")
     }
-    return(jags.out)
+  } else if (overdispersed == FALSE) {
+    ## chains fixed at data mean - used if above code does not converge
+    ## invalidates assumptions about convergence, e.g. Gelman-Rubin diagnostic
+    j.inits <- function(chain) { list(beta.o = mean(data$Y)) }
   }
-
+  
+  j.model <- jags.model(file = jag.model.file, 
+                        data = data, 
+                        inits = j.inits,
+                        n.chains = j.chains)
+  
+  jags.out <- coda.samples(model = j.model, 
+                           variable.names = vars,
+                           n.iter = j.iter, 
+                           thin = max(c(2, j.iter / (5000 * 2))))
+  
+  ## I would have done a while loop, but it could take forever 
+  ## So just give one chance to try again
+  if (gelman.diag(jags.out)$mpsrf > 1.2) {
+    logger.warn("model did not converge; re-running with j.iter * 10")
+    jags.out <- coda.samples(model = j.model, 
+                             variable.names = vars,
+                             n.iter = j.iter * 10, 
+                             thin = max(c(2, j.iter / (500 * 2))))
```

Comment:
Some of the more recent Bayesian stuff I've read advises against thinning in favor of calculating an effective sample size. Should we change this accordingly...?